qiskit_nature/drivers/second_quantization/gaussiand/gaussiandriver.py

```diff
@@ -129,6 +131,19 @@ def check_installed() -> None:
         """
         check_valid()
 
+    @staticmethod
+    def check_method_supported(method: MethodType) -> None:
+        """
+        Checks that Gaussian supports this method.
+        Args:
+            method: Method type
+
+        Raises:
+            NotSupportedMethodError: If method not supported.
+        """
+        if method in [MethodType.RKS, MethodType.ROKS, MethodType.UKS]:
+            raise NotSupportedMethodError(f"Invalid Gaussian method {method.value}.")
```

Comment:
I am wondering what the more scalable approach is. Although I find it unlikely that there will be many more methods added in the future, _if_ we do add another one, in this current implementation you would have to go into each driver and add the new method to this list (because most likely most drivers won't have it).
Thus, it may be favorable to check the method which _are_ supported and raise an error if the current method is not in that set. In this way, future updates are likely fewer in number :thinking: 

The same can obviously be said for all other drivers, too.

Reply:
Changed.